### ISSUE TITLE
Source Map is cached

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#7614]
CC @bvaughn

It looks like Firefox is caching the source maps after the first eval() 🤔

https://github.com/babel/website/pull/1315#issuecomment-323532189

**Comment by jasonLaster** [comment]
Hey @bvaughn would you be able to re-try this?

**Comment by jasonLaster** [comment]
@bvaughn break on exceptions should be in the breakpoints panel :)

sorry for the parity change

**Comment by bvaughn** [comment]
Ah, I see. It moved.

I wouldn't say the experience is ideal.

1. Open [this page](https://babeljs.io/en/repl#?babili=true&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEBiD29oG8BQ1rHgOwgFwCcBXYPeAgCjAEoV0NowLqBuegX1U9QDMjtSASxzQARmEo06GbAFMA7nERVWXVKgD0G6ABFZIWXlnQ8AC0EwQguanGUAjKyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cbabili&prettier=true&targets=&version=6.26.0&envVersion=)
2. Enable breakpoints for caught exceptions
3. Modify the code (delete the comment line) and then look at where the debugger stops.

Compare Chrome and Firefox behaviors.

Chrome shows code _after_ compilation– not ideal but...
![screen shot 2019-01-07 at 1 39 54 pm](https://user-images.githubusercontent.com/29597/50795023-ccf1f180-1281-11e9-9338-6e4ab6bee6b9.png)

Firefox shows the REPL code itself:
![screen shot 2019-01-07 at 1 41 04 pm](https://user-images.githubusercontent.com/29597/50795077-ebf08380-1281-11e9-81f4-f9f18c6743a4.png)
![screen shot 2019-01-07 at 1 41 10 pm](https://user-images.githubusercontent.com/29597/50795078-ebf08380-1281-11e9-82ed-93670873e624.png)

**Comment by jasonLaster** [comment]
interesting. thanks

can you give me a bit of an overview of what you would expect and why?

**Comment by bvaughn** [comment]
Ideally, Safari's behavior– showing the error in the context of the original code (the source map):
![screen shot 2019-01-07 at 2 14 25 pm](https://user-images.githubusercontent.com/29597/50796485-97033c00-1286-11e9-9e3d-2cab1f76a655.png)

**Comment by jasonLaster** [comment]
I see, i was confused because it looked like we were paused before the eval... 

the issue is that this code in the REPL has an exception!
<img width="415" alt="screen shot 2019-01-07 at 5 33 46 pm" src="https://user-images.githubusercontent.com/254562/50797347-78ab3980-12a2-11e9-8c18-7476873d43f4.png">

Got it, this makes more sense. thanks!

**Comment by bvaughn** [comment]
Oh, sorry. Yeah– in this case, I wrote some code with an intentional error– enabled the "Evaluate" option of the REPL (so that it runs the code after compilation)– and then looked at the breakpoint.

My apologies for leaving out that context 😅

**Comment by jasonLaster** [comment]
No worries, here are some steps for reproducing in safari

1. go. to preferences > advanced > enable developer tools
2. in the debugger top left select pause on assertions, all exceptions

<img width="1320" alt="screen shot 2019-01-07 at 5 37 36 pm" src="https://user-images.githubusercontent.com/254562/50797526-1acb2180-12a3-11e9-8e6b-812ac3418bd9.png">

**Comment by jasonLaster** [comment]
The interesting thing with this bug is that the client thinks it is paused in `scopedEval.js` and  `scopedEval.js` does not have a source map. For some reason, we're not paused in the evaled code, which is visible in the `code` preview...

<img width="760" alt="screen shot 2019-01-07 at 5 40 56 pm" src="https://user-images.githubusercontent.com/254562/50797654-79909b00-12a3-11e9-934e-345d19e40868.png">

---
<img width="755" alt="screen shot 2019-01-07 at 5 40 48 pm" src="https://user-images.githubusercontent.com/254562/50797655-79909b00-12a3-11e9-9597-b035e151f9a7.png">

---

<img width="460" alt="screen shot 2019-01-07 at 5 42 31 pm" src="https://user-images.githubusercontent.com/254562/50797710-b0ff4780-12a3-11e9-9d6c-4ab331d90421.png">

**Comment by jasonLaster** [comment]
The strangest thing is that the pause reason suggests we're evaluating the evaled script

the pause reason is `TypeError: c is not a function`, which is coming from the generated code in the eval'd context

<img width="986" alt="screen shot 2019-01-07 at 5 46 25 pm" src="https://user-images.githubusercontent.com/254562/50797889-51556c00-12a4-11e9-9dbd-e1aade25db37.png">

**Comment by jasonLaster** [comment]
Okay, after digging a bit more into it it looks like our `threadClient.getFrames` function fails to get the full exception stack

the exception stack has the 3 needed frames that are missing from `getFrames`

<img width="590" alt="screen shot 2019-01-07 at 6 15 41 pm" src="https://user-images.githubusercontent.com/254562/50799008-659b6800-12a8-11e9-821d-1815a0f61316.png">
<img width="975" alt="screen shot 2019-01-07 at 6 15 37 pm" src="https://user-images.githubusercontent.com/254562/50799009-659b6800-12a8-11e9-89ba-b2630345222a.png">

**Comment by bvaughn** [comment]
Yay! :)

**Comment by bhackett1024** [comment]
Thanks for diagnosing this, Logan.  I filed [bug 1518961](https://bugzilla.mozilla.org/show_bug.cgi?id=1518961) to fix this.  The logic for that !url check was added in [bug 1448431](https://bugzilla.mozilla.org/show_bug.cgi?id=1448431); the fix changes the mechanism for that check so that we ignore exceptions in console evaluations but not in normal debuggee evaluations.

With this fix we pause in the generated code at the point the exception is thrown (the same as what Chrome does).  I'll see if fixing #7682 causes us to pause in the original code instead.
